functions/index.js

```diff
@@ -1,8 +1,215 @@
+    	console.log("Successfully created new user:", userRecord.uid);
+    	return res.status(200).json(newUser);
+ 	 })
+  	 .catch(function(error) {
+    	console.log("Error creating new user:", error);
+    	return res.status(400).send(error);
+  	 });
+});
+
+/************************************************************************
+* Author: Jacob Nogle ------------------------------------------ 1/2018 *
+* Delete a specified user                                               *
+* Expects uid in query params                                           *
+*************************************************************************/
+app.delete('/users/delete-user/:uid', (req, res) => {
+    const uid = req.params.uid;
+    admin.auth().deleteUser(uid)
+        .then(function() {
+            console.log("Successfully deleted user");
+            return res.status(200).send();
+        })
+        .catch(function(error) {
+            console.log("Error deleting user:", error);
+            return res.status(400).send(error);
+        })
+});
+
+exports.api = functions.https.onRequest(app);
+
+/*******************************************************************
+* Author: Jacob Nogle ------------------------------------- 1/2018 *
+* Function to write newly created users to database                *
+* Triggered by new user creation                                   *
+********************************************************************/
+exports.storeNewUser = functions.auth.user().onCreate(event => {
+	const uid = event.data.uid;
+	const email = event.data.email;
+	const phoneNumber = event.data.phoneNumber;
+	const name = event.data.displayName;
+    const photoURL = event.data.photoURL;
+    
+	const coll = db.collection("users");
+    //Add default data 
+	return coll.doc(uid).set({
+		email : email,
+		phoneNumber : phoneNumber,
+		name: name,
+        profile: {
+            isPublic: true,
+            isStaff: false,
+            isCommunityGroupLeader: false,
+            isMinistryTeamLeader: false,
+            isSummerMissionLeader: false
+        },
+        notifications: {
+            ministryTeamUpdates: true,
+            communityGroupUpdates: true,
+            summerMissionUpdates: true
+        },
+        permissions: {
+            isAdmin: false, 
+            isVerified: false 
+        },
+        lastActive: Date.now()
+	})
+	.then(function() {
+		console.log("New user successfully added to database");
+	})
+	.catch(function(error) {
+		console.error("Error adding new user to database: ", error);
+	});
+});
+
+/*******************************************************************
+* Author: Jacob Nogle ------------------------------------- 1/2018 *
+* Function to remove a user from the database                      *
+* Triggered by user deletion                                       *
+********************************************************************/
+exports.removeUser = functions.auth.user().onDelete(event => {
```

Comment:
I'm a little confused as to the purpose of this function and how it differs from the delete function above.